File: reference/php/Internal/Node.php
<?php

namespace DistributedHashMap\Internal;

use Dapr\Deserialization\Attributes\ArrayOf;
use Dapr\Serialization\Attributes\AlwaysObject;

class Node
{
    /**
     * Node constructor.
     *
     * @param string[] $items
     * @param KeyTrigger[] $triggers
     */
    public function __construct(
        #[ArrayOf('string')]
        public array $items = [],

        #[ArrayOf(KeyTrigger::class)]
        #[AlwaysObject]
        public array $triggers = []
    ) {
    }
}

File: reference/csharp/DistributedHashMap/DistributedHashMap/Internal/Node.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace DistributedHashMap.Internal
{
    internal class Node
    {
        [JsonPropertyName("items")]
        public Dictionary<string, string> Items { get; set; } = new Dictionary<string, string>();
    }
}
